code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/PushTemplateImageUtil.kt

```diff
@@ -83,10 +53,105 @@ internal object PushTemplateImageUtil {
      *
      * @param cacheService the AEPSDK [CacheService] to use for caching or retrieving
      * downloaded image assets
-     * @param uri [String] containing an image asset url
+     * @param uriList [String] containing an image asset url
      * @return [Bitmap] containing the image referenced by the `String` uri
      */
-    internal fun downloadImage(cacheService: CacheService, uri: String?): Bitmap? {
+    internal fun downloadImage(
+        cacheService: CacheService,
+        uriList: List<String?>
+    ): Int {
+        val assetCacheLocation = getAssetCacheLocation()
+        if (assetCacheLocation.isNullOrEmpty() || uriList.isEmpty()) {
```

Comment:
we can return before getting the asset cache location if uriList.isNullOrEmpty()